scripts/assets/file_content_validation_checks.textproto

```diff
@@ -408,3 +408,13 @@ file_content_checks {
   exempted_file_name: "app/src/main/java/org/oppia/android/app/application/AbstractOppiaApplication.kt"
   exempted_file_patterns: ".+?Test\\.kt"
 }
+file_content_checks {
+  file_path_regex: ".+?.kt"
+  prohibited_content_regex: ".+?\\.(post|postDelayed)\\(.+?"
+  failure_message: "Using post() and postDelayed() method is prohibited as it can lead to NullPointerException errors. Use doOnPreDraw or doOnLayout instead. Checkout this resource for a better understanding of the issue: https://betterprogramming.pub/stop-using-post-postdelayed-in-your-android-views-9d1c8eeaadf2"
```

Comment:
```suggestion
  failure_message: "Prefer avoiding post() and postDelayed() method as they can can lead to subtle and difficult-to-debug crashes. Prefer using LifecycleSafeTimerFactory for most cases when an operation needs to run at a future time. For cases when state needs to be synchronized with a view, use doOnPreDraw or doOnLayout instead. For more context on the underlying issue, see: https://betterprogramming.pub/stop-using-post-postdelayed-in-your-android-views-9d1c8eeaadf2."
```

Softened up the language a bit, and added to the recommendation per Oppia best practices.

Reply:
This has been applied in the latest changes.